sdk/src/fee.rs

```diff
@@ -78,52 +96,45 @@ impl FeeStructure {
             BASE_CONGESTION / current_congestion
         };
 
-        let mut compute_budget = ComputeBudget::default();
```

Comment:
can we save this refactor for a separate PR?  i think it's substantial enough to stand on its own.  all of those feature gate bools being removed from the signature scared me at first :joy:

Reply:
We cannot access `ComputeBudget` from `sdk`, since it's in `program-runtime/src`. So this refactor is necessary to be part of this PR.

Reply:
it should be doable in a preliminary pr, then?

Reply:
Created this PR to split this part of the change. https://github.com/solana-labs/solana/pull/32487